# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCDTMFSender.h
# Commit: 0991ebfeb365
# Full Hash: 0991ebfeb3657ca4e5c379c5211f6f4d1e29968d
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Make RTCDTMFSender signal DTMF tones through a MediaEvent. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112859
# ==============================================================================

diff -r bd3db9010131 -r 0991ebfeb365 dom/media/webrtc/jsapi/RTCDTMFSender.h
--- a/dom/media/webrtc/jsapi/RTCDTMFSender.h	Sun Apr 18 21:47:29 2021 +0200
+++ b/dom/media/webrtc/jsapi/RTCDTMFSender.h	Sun Apr 18 21:06:49 2021 +0200
@@ -5,6 +5,7 @@
 #ifndef _RTCDTMFSender_h_
 #define _RTCDTMFSender_h_
 
+#include "MediaEventSource.h"
 #include "mozilla/DOMEventTargetHelper.h"
 #include "mozilla/RefPtr.h"
 #include "js/RootingAPI.h"
@@ -17,15 +18,26 @@
 class AudioSessionConduit;
 class TransceiverImpl;
 
+struct DtmfEvent {
+  DtmfEvent(int aPayloadType, int aPayloadFrequency, int aEventCode,
+            int aLengthMs)
+      : mPayloadType(aPayloadType),
+        mPayloadFrequency(aPayloadFrequency),
+        mEventCode(aEventCode),
+        mLengthMs(aLengthMs) {}
+  const int mPayloadType;
+  const int mPayloadFrequency;
+  const int mEventCode;
+  const int mLengthMs;
+};
+
 namespace dom {
 
 class RTCDTMFSender : public DOMEventTargetHelper,
                       public nsITimerCallback,
                       public nsINamed {
  public:
-  explicit RTCDTMFSender(nsPIDOMWindowInner* aWindow,
-                         TransceiverImpl* aTransceiver,
-                         AudioSessionConduit* aConduit);
+  RTCDTMFSender(nsPIDOMWindowInner* aWindow, TransceiverImpl* aTransceiver);
 
   // nsISupports
   NS_DECL_NSITIMERCALLBACK
@@ -36,6 +48,7 @@
   // webidl
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
+  void SetPayloadType(int32_t aPayloadType, int32_t aPayloadFrequency);
   void InsertDTMF(const nsAString& aTones, uint32_t aDuration,
                   uint32_t aInterToneGap, ErrorResult& aRv);
   void GetToneBuffer(nsAString& aOutToneBuffer);
@@ -43,13 +56,17 @@
 
   void StopPlayout();
 
+  MediaEventSource<DtmfEvent>& OnDtmfEvent() { return mDtmfEvent; }
+
  private:
   virtual ~RTCDTMFSender() = default;
 
   void StartPlayout(uint32_t aDelay);
 
   RefPtr<TransceiverImpl> mTransceiver;
-  RefPtr<AudioSessionConduit> mConduit;
+  MediaEventProducer<DtmfEvent> mDtmfEvent;
+  Maybe<int32_t> mPayloadType;
+  Maybe<int32_t> mPayloadFrequency;
   nsString mToneBuffer;
   uint32_t mDuration = 0;
   uint32_t mInterToneGap = 0;